src/compas/scene/scene.py

```diff
@@ -43,47 +44,51 @@ class Scene(Tree):
     @property
     def __data__(self):
         # type: () -> dict
-        items = {str(object.item.guid): object.item for object in self.objects if object.item is not None}
         return {
             "name": self.name,
-            "root": self.root.__data__,  # type: ignore
-            "items": list(items.values()),
+            "attributes": self.attributes,
+            "datastore": self.datastore,
+            "objectstore": self.objectstore,
+            "tree": self.tree,
         }
 
-    @classmethod
-    def __from_data__(cls, data):
-        # type: (dict) -> Scene
-        scene = cls(data["name"])
-        items = {str(item.guid): item for item in data["items"]}
-
-        def add(node, parent, items):
-            for child_node in node.get("children", []):
-                settings = child_node["settings"]
-                if "item" in child_node:
-                    guid = child_node["item"]
-                    sceneobject = parent.add(items[guid], **settings)
-                else:
-                    sceneobject = parent.add(Group(**settings))
-                add(child_node, sceneobject, items)
-
-        add(data["root"], scene, items)
-
-        return scene
-
-    def __init__(self, name="Scene", context=None):
-        # type: (str, str | None) -> None
-        super(Scene, self).__init__(name=name)
-        super(Scene, self).add(TreeNode(name="ROOT"))
+    def __init__(self, context=None, datastore=None, objectstore=None, tree=None, **kwargs):
+        # type: (str | None, dict | None, dict | None, Tree | None, **kwargs) -> None
+        super(Scene, self).__init__(**kwargs)
+
         self.context = context or detect_current_context()
+        self.datastore = datastore or {}
+        self.objectstore = objectstore or {}
+        self.tree = tree or Tree()
+        if self.tree.root is None:
+            self.tree.add(TreeNode(name=self.name))
+
+    def __repr__(self):
+        # type: () -> str
+
+        def node_repr(node):
+            # type: (TreeNode) -> str
+            if node.is_root:
+                return node.name
+            else:
+                sceneobject = self.objectstore[node.name]
+                return str(sceneobject)
+
+        return self.tree.get_hierarchy_string(node_repr=node_repr)
+
+    @property
+    def items(self):
+        # type: () -> list[compas.data.Data]
+        return list(self.datastore.values())
 
     @property
     def objects(self):
         # type: () -> list[SceneObject]
-        return [node for node in self.nodes if not node.is_root]  # type: ignore
+        return list(self.objectstore.values())
 
     @property
     def context_objects(self):
-        # type: () -> list
+        # type: () -> list[SceneObject]
```

Comment:
these are not scene objects but the guids of the objects created as representation in the visualisation context